### ISSUE TITLE
Self hosted Sentry - A required debug information file was missing.

### ISSUE THREAD

**Issue by DevMobileAS** [getsentry/self-hosted#2632]
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Steps to Reproduce

We're running two separate instances of the sentry backend - both with the same issue.

After uploading a dsym file for an iOS project it's successfully displayed in the project settings:
<img width="1175" alt="Bildschirmfoto 2023-12-06 um 13 50 43" src="https://github.com/getsentry/sentry/assets/75465811/9d56c3eb-1fa4-4b9e-85db-5274ef031a77">

But the crashlog for the regarding iOS App does still not symbolicate correctly:
<img width="1942" alt="Bildschirmfoto 2023-12-06 um 13 53 48" src="https://github.com/getsentry/sentry/assets/75465811/4fc83ee0-c35b-4250-b30d-0ef201ef40f6">
<img width="1953" alt="Bildschirmfoto 2023-12-06 um 13 53 54" src="https://github.com/getsentry/sentry/assets/75465811/c14d6f9e-6493-4f67-a3c9-c136cef5033c">

The "Images Loaded" section looks like this:
<img width="1960" alt="Bildschirmfoto 2023-12-06 um 13 54 34" src="https://github.com/getsentry/sentry/assets/75465811/347e4841-e0ba-4055-8cc1-2d5a207a5b45">
<img width="1832" alt="Bildschirmfoto 2023-12-06 um 13 55 36" src="https://github.com/getsentry/sentry/assets/75465811/298d30e6-064d-4caa-a035-a5fcaeab7925">

The issue was definitely sent hours after uploading the dsym (initially I also got another the warning, that dsym was only recently uploaded, so I waited the night and tried again, with same result)

----

So I've turned on 'Reprocessing' in the settings to check what's wrong. Result is this:
<img width="2285" alt="Bildschirmfoto 2023-12-06 um 13 53 03" src="https://github.com/getsentry/sentry/assets/75465811/12510edf-dde9-4b93-80d3-e4df019c2cf4">

<img width="1191" alt="Bildschirmfoto 2023-12-06 um 13 52 24" src="https://github.com/getsentry/sentry/assets/75465811/1c1509e2-4b17-4bf7-8ca4-493ce26829b6">

I also checked the dsym itself on my Mac:
```
$ sentry-cli difutil check /Users/yi01144/Library/Developer/Xcode/Archives/2023-12-05/VMA\ 05.12.23\,\ 17.58.xcarchive/dSYMs/VMA.app.dSYM/Contents/Resources/DWARF/VMA 
Debug Info File Check
  Type: dsym debug companion
  Contained debug identifiers:
    > Debug ID: 7a851641-f701-3c0c-8c61-093cda3f348f
      Code ID:  7a851641f7013c0c8c61093cda3f348f
      Arch:     arm64
  Contained debug information:
    > symtab, debug, unwind
  Usable: yes
```

The "debug ID" is exactly the same here, so I'm lost why the error is given and it still can't find the dsym which is clearly there and valid.

I really hope that you have any idea for further debugging or what's causing the issue.

Kind regards
Björn

### Expected Result

I really would like that sentry finds and uses the dsym to symbolicate my crash logs.

### Actual Result

Error finding dsym which are imho available.

### Product Area

Issues

### Version

23.11.2

**Comment by hubertdeng123** [comment]
Are there any logs in symbolicator that may show an issue that is occurring?

**Comment by chris-laack** [comment]
Hi @hubertdeng123,

Thank you very much for your feedback 

the first thing we did today was to adjust the configuration of symbolicator_service -> INTERNAL_SYSTEM_IPS. The error "ERROR: Failed to fetch file list from 
sentry: failed to fetch data from Sentry
caused by: bad status code from Sentry: 401 Unauthorized" is now fixed. 
However, we now get the following error from the 
symbolicator_service: 

2023-12-11T14:41:42.020995585Z WARN symbolicator_service::download::sentry: Sentry API returned status code 401 Unauthorized
2023-12-11T14:41:42.021060197Z ERROR symbolicator_service::download: Failed to fetch file list error=download failed: 401 Unauthorized

Unfortunately, I can't find anything that goes wrong here.

**Comment by azaslavsky** [comment]
Do you have logs on the Sentry side when this 401 occurs?

**Comment by chris-laack** [comment]
I would now need some support. From which container do I need the logs? Sentry worker or sentry web?

**Comment by azaslavsky** [comment]
It seems like it is pinging Sentry web, though I concede that I am not a symbolicator expert.

**Comment by chris-laack** [comment]
I think I have found the correct log in the sentry web container. 

`Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sentry/../sentry_sdk/integrations/django/__init__.py", line 500, in wsgi_request_event_processor
    _set_user_info(request, event)
  File "/usr/local/lib/python3.8/site-packages/sentry/../sentry_sdk/integrations/django/__init__.py", line 576, in _set_user_info
    user = getattr(request, "user", None)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/request.py", line 227, in user
    self._authenticate()
  File "/usr/local/lib/python3.8/site-packages/rest_framework/request.py", line 380, in _authenticate
    user_auth_tuple = authenticator.authenticate(self)
  File "/usr/local/lib/python3.8/site-packages/sentry/api/authentication.py", line 184, in authenticate
    return self.authenticate_token(request, force_str(auth[1]))
  File "/usr/local/lib/python3.8/site-packages/sentry/api/authentication.py", line 336, in authenticate_token
    raise AuthenticationFailed("Invalid token")
rest_framework.exceptions.AuthenticationFailed: Invalid token
14:41:40 [ERROR] sentry_sdk.errors: Internal error in sentry_sdk
14:41:40 [ERROR] sentry.auth.system: Trying to use `SystemToken` from non-internal IP
14:41:40 [INFO] sentry.access.api: api.access (method='GET' view='sentry.api.endpoints.debug_files.DebugFilesEndpoint' response=401 user_id='None' is_app='None' token_type='None' is_frontend_request='False' organization_id='None' auth_id='None' path='/api/0/projects/aok-systems/vma-ios/files/dsyms/' caller_ip='172.18.40.10' user_agent='symbolicator/23.11.2' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.32301783561706543 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
14:41:40 [INFO] sentry.access.api: api.access (method='GET' view='sentry.api.endpoints.debug_files.DebugFilesEndpoint' response=401 user_id='None' is_app='None' token_type='None' is_frontend_request='False' organization_id='None' auth_id='None' path='/api/0/projects/aok-systems/vma-ios/files/dsyms/' caller_ip='172.18.40.10' user_agent='symbolicator/23.11.2' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.1944866180419922 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
14:41:40 [ERROR] sentry.auth.system: Trying to use `SystemToken` from non-internal IP
14:41:40 [WARNING] django.request: Unauthorized: /api/0/projects/aok-systems/vma-ios/files/dsyms/ (status_code=401 request=<WSGIRequest: GET '/api/0/projects/aok-systems/vma-ios/files/dsyms/?_last_upload=1702305634335&debug_id=e829a395-21f6-3f8e-bd6d-8729b633fb93'>)
14:41:40 [ERROR] sentry.auth.system: Trying to use `SystemToken` from non-internal IP
14:41:40 [WARNING] django.request: Unauthorized: /api/0/projects/aok-systems/vma-ios/files/dsyms/ (status_code=401 request=<WSGIRequest: GET '/api/0/projects/aok-systems/vma-ios/files/dsyms/?_last_upload=1702305634335&debug_id=2b3c63df-8688-3dde-8dea-66fc7fc42404'>)
14:41:40 [INFO] sentry.access.api: api.access (method='POST' view='sentry.api.endpoints.relay.project_configs.RelayProjectConfigsEndpoint' response=200 user_id='None' is_app='None' token_type='None' is_frontend_request='False' organization_id='None' auth_id='None' path='/api/0/relays/projectconfigs/' caller_ip='172.21.0.49' user_agent='None' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.08039593696594238 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
14:41:40 [INFO] sentry.access.api: api.access (method='GET' view='sentry.api.endpoints.debug_files.DebugFilesEndpoint' response=401 user_id='None' is_app='None' token_type='None' is_frontend_request='False' organization_id='None' auth_id='None' path='/api/0/projects/aok-systems/vma-ios/files/dsyms/' caller_ip='172.18.40.10' user_agent='symbolicator/23.11.2' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.16688251495361328 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
14:41:40 [INFO] sentry.access.api: api.access (method='GET' view='sentry.api.endpoints.debug_files.DebugFilesEndpoint' response=401 user_id='None' is_app='None' token_type='None' is_frontend_request='False' organization_id='None' auth_id='None' path='/api/0/projects/aok-systems/vma-ios/files/dsyms/' caller_ip='172.18.40.10' user_agent='symbolicator/23.11.2' rate_limited='False' rate_limit_category='None' request_duration_seconds=0.1794741153717041 rate_limit_type='DNE' concurrent_limit='None' concurrent_requests='None' reset_time='None' group='None' limit='None' remaining='None')
14:41:40 [WARNING] django.request: Unauthorized: /api/0/projects/aok-systems/vma-ios/files/dsyms/ (status_code=401 request=<WSGIRequest: GET '/api/0/projects/aok-systems/vma-ios/files/dsyms/?_last_upload=1702305634335&debug_id=dfed7494-ed51-3674-abc0-0407f7292973'>)
14:41:40 [WARNING] django.request: Unauthorized: /api/0/projects/aok-systems/vma-ios/files/dsyms/ (status_code=401 request=<WSGIRequest: GET '/api/0/projects/aok-systems/vma-ios/files/dsyms/?_last_upload=1702305634335&debug_id=7a851641-f701-3c0c-8c61-093cda3f348f'>)
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sentry/api/authentication.py", line 331, in authenticate_token
    ApiToken.objects.filter(token=token_str)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 435, in get
    raise self.model.DoesNotExist(
sentry.models.apitoken.ApiToken.DoesNotExist: ApiToken matching query does not exist.`

**Comment by azaslavsky** [comment]
`14:41:40 [ERROR] sentry.auth.system: Trying to useSystemToken from non-internal IP` seems to indicate that your symbolicator instance is running on a different IP than what is expected.

Reading [this comment](https://github.com/getsentry/sentry/blob/17b7a7920feb0d75df65f37e4a0acb2353600310/src/sentry/auth/system.py#L62) seems to indicate you need to adjust your IP ranges to include the symbolicator server. Some helpful info that's worth grabbing to debug this:

- What is the IP of your `symbolicator` container?
- What is the IP of your `web` container?
- What does the value of your [INTERNAL_SYSTEM_IPS](https://github.com/getsentry/self-hosted/blob/76ab9a596e7dc32c468f387dd45ba637e8cfad53/sentry/sentry.conf.example.py#L40) resolve to at runtime?

Basically, you'll need to ensure that `INTERNAL_SYSTEM_IPS` includes the IP of your symbolicator instance.

**Comment by chris-laack** [comment]
Hi,
the following ip addresses are used:
Symbolicator:
docker inspect \
  -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' sentry-self-hosted-symbolicator-1
172.24.0.9

web:
docker inspect   -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' sentry-self-hosted-web-1
172.24.0.34

Internal Server IP´s in the sentry.conf.py file: 

INTERNAL_SYSTEM_IPS = (
    
    "10.100.0.0/16",
    "172.0.0.0/8",
)
the ip range i had at first was unfortunately wrong, because it was not used by docker itself. 
Unfortunately, I can't have the current setting tested because my colleague is on vacation until the beginning of January. 
Is it possible to leave the ticket open until then?

**Comment by azaslavsky** [comment]
The tickets auto-close after 3 weeks - if you want the bug to stay open, just add a comment before then. Good luck with the IP adjustments!

**Comment by chris-laack** [comment]
3 weeks is enough. Thanks for the info
Thanks

**Comment by chris-laack** [comment]
Hi, 

I wish you a happy new year. 
Our problem is solved with the settings. Our developers were able to test everything successfully. 
I would like to thank you for your support. 
The ticket can be closed. 

Chris

**Comment by DevMobileAS** [comment]
Yea, the changes Chris recently applied were successful and our strack traces could finally be symbolicated. Yea!

Huge thanks to you all and happy new year!

PS: Still the error shown in the screenshots in the first post were wrong about "This issue was processed before this debug information file was available". So maybe this should be fixed and instead log a connection issue to the symbolicator service to prevent other users from running into that issue?
